native/NativeSampleApps/SmartSyncExplorer/src/com/salesforce/samples/smartsyncexplorer/ui/MainActivity.java

```diff
@@ -286,7 +291,8 @@ private void refreshList() {
     }
 
 	private void refreshList(List<ContactObject> data) {
-		listAdapter.setData(data);
+		// NB: We feed the data to nameFilter, and in turns it feeds the (filtered) data to listAdapter
+		nameFilter.setData(data);
```

Comment:
The NameFieldFilter remembers the list of ContactObjects, setting that list causes the NameFieldFilter to update the ContactListAdapter. We don't change that list on the ContactListAdapter directly anymore.